dlx_rest/api/__init__.py

```diff
@@ -859,6 +859,53 @@ def get(self, collection, record_id):
         lock_status= item_locked(collection, record_id)
         return lock_status, 200
 
+# Record parse
+@ns.route('/marc/<string:collection>/parse')
+@ns.param('collection', '"bibs" or "auths"')
+class RecordParse(Resource):
+    args = reqparse.RequestParser()
+    args.add_argument(
+        'format',
+        required=True,
+        type=str, 
+        choices=['xml', 'mrk', 'csv']
+    )
+
+    @ns.doc(description='Parse the string and return a native JSON representation')
+    @ns.expect(args)
+    def get(self, collection):
```

Comment:
I suggest changing the method from GET to POST. Per the HTTP specification a GET request puts all of the data in the URL and can't have a Body, so for sufficiently long records, the parse endpoint will either fail or cut off data. I'm looking for more specific answers on the max URL length for a Python request, but I'm seeing maybe 8000 characters.